src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -190,7 +196,11 @@ protected virtual void Dispose(bool disposing)
                     foreach (var pair in _workerStates)
                     {
                         // TODO #3296 - send WorkerTerminate message to shut down language worker process gracefully (instead of just a killing)
-                        pair.Value.Channel.Dispose();
+                        // WebhostLanguageWorkerChannels life time is managed by LanguageWorkerChannelManager
+                        if (!pair.Value.Channel.IsWebhostChannel)
+                        {
+                            pair.Value.Channel.Dispose();
```

Comment:
Placeholders for Java were because, when ping warmup C# site, workerState is not created in the function dispatcher.  This bug was specifically affecting deployment scenarios when there are multiple jobhost restarts

Reply:
Ideally, this would be a little cleaner and not require knowledge from consumers to know what to dispose (or a property to indicate the scope). Wouldn't block on this now, but let's plan on that work.

Reply:
I agree. Opened https://github.com/Azure/azure-functions-host/issues/4107